packages/runtime/container-runtime/src/pendingStateManager.ts

```diff
@@ -106,6 +106,9 @@ export class PendingStateManager implements IDisposable {
     public getLocalState(): IPendingLocalState | undefined {
         assert(this.initialStates.isEmpty(), 0x2e9 /* "Must call getLocalState() after applying initial states" */);
         if (this.hasPendingMessages()) {
+            while (this.pendingStates.peekFront()?.type !== "message") {
```

Comment:
I would try to fix this bug in some other way than adding a side effect to a function which is supposed to get something..